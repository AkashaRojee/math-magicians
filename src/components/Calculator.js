import React, { Component } from 'react';
import Display from './Display';
import Keyboard from './Keyboard';
import calculate from '../logic/calculate';
import DataObject from '../objects/DataObject';
// import operate from '../logic/operate';

class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = {
      display: '0',
      // dataObject: new DataObject(),
    };
    this.dataObject = new DataObject();
  }

  handleClick(key) {
    this.dataObject = Object.assign(this.dataObject, calculate(this.dataObject, key));
    this.setState({
      display:
        `${(this.dataObject.total || this.dataObject.next)
            + (this.dataObject.operation || '')
            + ((this.dataObject.operation && this.dataObject.next) || '')}`,
    });
    console.log(this.dataObject);
  }

  render() {
    const { display } = this.state;

    return (
      <div className="calculator">
        <Display value={display} />
        <Keyboard onKeyboardClick={(key) => this.handleClick(key)} />
      </div>
    );
  }
}

export default Calculator;
