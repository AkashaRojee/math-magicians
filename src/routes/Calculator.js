/* eslint-disable */

const Calculator = () => {
  return (
    <span>calculator</span>
  );
}

export default Calculator;