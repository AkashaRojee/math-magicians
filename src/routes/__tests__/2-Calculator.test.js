import renderer from 'react-test-renderer';
import Calculator from '../2-Calculator';

test('Calculator route renders correctly', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
