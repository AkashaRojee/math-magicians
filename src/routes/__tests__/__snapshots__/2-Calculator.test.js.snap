// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator route renders correctly 1`] = `
<main
  className="calculator-route"
>
  <h2>
    Let's do some math!
  </h2>
  <div
    className="calculator"
  >
    <span
      className="display"
    >
      
      0
    </span>
    <div
      className="keyboard"
    >
      <button
        className="bg-purple"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="bg-purple-light"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="bg-purple-light"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="bg-purple-light"
        onClick={[Function]}
        type="button"
      >
        /
      </button>
      <button
        className="bg-purple-dark"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="bg-purple-dark"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="bg-purple-dark"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="bg-purple-light"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="bg-purple-dark"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="bg-purple-dark"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="bg-purple-dark"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="bg-purple-light"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="bg-purple-dark"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="bg-purple-dark"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="bg-purple-dark"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="bg-purple-light"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="span-two-col bg-purple-dark"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="bg-purple-dark"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="bg-cyan"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</main>
`;
